### PR TITLE
[Pages] Implement virtual Homepage actions

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #18371 

This implements the missing part of the #18400 PR, which are the callbacks and actions to:
- Open the SiteEditor on a WebView (injecting CSS to hide some navigation elements of the page)
- Open the support page about `Templates` on the external browser

This also implements some code to inject CSS to the page being loaded in the WebView.

## To test

Requirements:
- At least 1 site using a block-based theme, such as `Vetro`, with the homepage set to display "your latest posts" (on the web you can find it at `Appearance > Customize > Homepage Settings`

1. Login with your account
2. Select a site with a block-based theme
3. Go to `Debug Settings`
4. Enable the `SiteEditorMVPFeatureConfig`
5. Go back to the `My Site` screen
6. Tap on `Pages`
7. **Verify** the "virtual" homepage item is shown
8. Tap the card
9. **Verify** the Site Editor page is opened on a WebView directly on the editor for the home template/page
10. **Verify** UI for navigating inside the editor is hidden (compare with web version on a browser)
11. Go back to the `Pages` screen
12. Tap the info icon
13. **Verify** it opens the support page about `Templates` on the external browser

If possible, test with different types of accounts, ex: WP.com account, self-hosted without Jetpack connection, and self-hosted with Jetpack connection. And also with different site types, ex: Simple sites and Atomic sites. And also WordPress and Jetpack apps.

## Regression Notes
1. Potential unintended areas of impact
Issues for other page types, issues opening the editor/external url.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Added some unit tests in the ViewModel responsible for those callbacks.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
**There were no UI changes, just UI interaction, so most actions below don't apply**
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [x] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java [hunk 1]
```diff
@@ -704,6 +747,15 @@ public void onWebViewPageLoaded() {
         refreshBackForwardNavButtons();
     }
 
+    private void injectCss(String css) {
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java]: 💡 Nitpick: would it make sense to add `@NonNull` and `final` to the `css` parameter?

**Review by RenanLukas — COMMENTED** [review]
Hey @thomashorta , thanks for the changes.

I've tested JP and WP, Atomic and simple sites, WP.com login and self-hosted. Everything worked as expected except for one scenario: WP app with a self-hosted site opened a different site when I tapped the virtual homepage button: it opened the WebView with the first site on my sites list selected (and not the SH I had selected).

**Comment by thomashorta** [comment]
> I've tested JP and WP, Atomic and simple sites, WP.com login and self-hosted. Everything worked as expected except for one scenario: WP app with a self-hosted site opened a different site when I tapped the virtual homepage button: it opened the WebView with the first site on my sites list selected (and not the SH I had selected).

Thanks for catching that. It seems that for jetpack-connected sites, the `isSelfHostedAdmin` field returns `false` even though the site IS self-hosted. In that case, the code was trying to use WPCom credentials, but that obviously don't work, so I changed the code to use other fields that more accurately check if the site is a `wpcom` site.

**Review by RenanLukas — APPROVED** [review]
LGTM :shipit:

**Pull request merged**